testing/sdk_repository.py

```diff
@@ -96,18 +100,194 @@ def remove_universe_repos(stub_urls):
+
+
+def add_dcos_files_to_registry(
+        temp_dir
+) -> None:
+    # Use DCOS_FILES_PATH if its set to a valid path OR use pytest's tmpdir.
+    dcos_files_path = os.environ.get('DCOS_FILES_PATH', '')
+    if not os.path.isdir(dcos_files_path):
+        dcos_files_path = str(temp_dir)
+    stub_universe_urls = get_universe_repos()
+    log.info('Using {} to build .dcos files (if not exists) from {}'.format(
+        dcos_files_path, stub_universe_urls
+    ))
+    dcos_files_list = build_dcos_files_from_stubs(
+        stub_universe_urls,
+        dcos_files_path,
+        temp_dir
+    )
+    log.info('Bundled .dcos files : {}'.format(dcos_files_list))
+    for file in dcos_files_list:
+        rc, out, err = sdk_cmd.run_raw_cli(' '.join([
+            'registry',
+            'add',
+            '--dcos-file={}'.format(file),
+            '--json'
+        ]))
+        assert rc == 0
+        assert len(json.loads(out)['packages']) > 0
+
+
+def build_dcos_files_from_stubs(
+        stub_universe_urls: list,
+        dcos_files_path: str,
+        temp_dir
+) -> list:
+    if not len(stub_universe_urls):
+        return stub_universe_urls
+    package_file_paths = []
+    for repo_url in stub_universe_urls:
+        headers = {
+            "User-Agent": "dcos/{}".format(sdk_cmd.dcos_version()),
+            "Accept": "application/vnd.dcos.universe.repo+json;"
+                      "charset=utf-8;version={}".format('v4'),
+        }
+        req = urllib.request.Request(repo_url, headers=headers)
+        with urllib.request.urlopen(req) as f:
+            data = json.loads(f.read().decode())
+            for package in data['packages']:
+                package_file_paths.append(
+                    build_dcos_file_from_universe_definition(
+                        package,
+                        dcos_files_path,
+                        temp_dir
+                    )
+                )
+    return package_file_paths
+
+
+def build_dcos_file_from_universe_definition(
+        package: dict,
+        dcos_files_path: str,
+        temp_dir
+) -> str:
+    # TODO Ideally we should `migrate` and then `build`.
+    del package['releaseVersion']
+    del package['selected']
+    package_json_file = temp_dir.mkdtemp().join(random_string())
+    package_json_file.write(json.dumps(package))
+    target = os.path.join(
+        dcos_files_path,
+        '{}-{}.dcos'.format(package['name'], package['version'])
+    )
+    if os.path.isfile(target):
+        log.info('Skipping build, using cached file : {}'.format(target))
+    else:
+        rc, _, _ = sdk_cmd.run_raw_cli(' '.join([
+            'registry',
+            'build',
+            '--build-definition-file={}'.format(str(package_json_file)),
+            '--output-directory={}'.format(dcos_files_path),
+            '--json'
+        ]))
+        assert rc == 0
+    assert os.path.isfile(target)
+    return target
+
+
+def grant_perms_for_registry_account(service_uid: str) -> None:
```

Comment:
It may be useful to have this in the `sdk_security` module instead?

Reply:
I tried to do that initially. But since the set of operations are specific to registry and all the `sdk_security` stuff use the `http` api, it didn't feel right to put this in there. We can always move this to `sdk_security` if/when we need to reuse a similar functionality.